src/coreclr/pal/src/arch/arm64/context2.S

```diff
@@ -104,6 +104,41 @@ LOCAL_LABEL(Done_CONTEXT_INTEGER):
     sub x0, x0,   CONTEXT_FLOAT_CONTROL_OFFSET + CONTEXT_NEON_OFFSET
 
 LOCAL_LABEL(Done_CONTEXT_FLOATING_POINT):
+    ldr w1, [x0, CONTEXT_ContextFlags]
+    // clangs assembler doesn't seem to support the mov Wx, imm32 yet
+    movz w2, #0x40, lsl #16
+    movk w2, #0x40
+    mov w3, w2
+    and w2, w1, w2
+    cmp w2, w3
+    b.ne LOCAL_LABEL(Done_CONTEXT_SVE)
+
+    ldr x1, [x0, CONTEXT_XSTATEFEATURESMASK_OFFSET]
+    tbz x1, #XSTATE_SVE_BIT, LOCAL_LABEL(Done_CONTEXT_SVE)
+
+    add x0, x0,   CONTEXT_SVE_OFFSET
+    str p0, [x0, CONTEXT_P0_VL, MUL VL]
```

Comment:
This data structure is hardcoded to 128 bit, but the VL register can be more than 128 bit. This will result into a buffer overrun on machines with >128 bit SVE.

Reply:
I have put up https://github.com/dotnet/runtime/pull/104174 to disable SVE if running on > 128 bit SVE

Reply:
https://github.com/dotnet/runtime/pull/104174 disables the SVE support in the JIT. I do not think it will prevent this PAL code from running on 256+ bit systems (correct me if I am missing some detail), so the buffer overrun is still a problem.

Reply:
My impression was that we check coreclr to see if SVE is enabled or not as opposed to if machine has SVE capabilities or not. But may be we are doing the later, so if we change it to former (use coreclr information if sve is enabled), then with my patch things should work.

Reply:
> I do not think it will prevent this PAL code from running on 256+ bit systems

Yes, your right.

I think all that needs happening is that in two places in context.cpp, 
`if (sve && sve->head.size >= SVE_SIG_CONTEXT_SIZE(sve_vq_from_vl(sve->vl)))`
needs changing to something like:
`if (sve && sve->head.size >= SVE_SIG_CONTEXT_SIZE(sve_vq_from_vl(sve->vl)) && SVE_IS_ENABLED_IN_CORECLR)`

That will prevent XSTATE_MASK_SVE from ever being set. Which will prevent any of the sve code in `context2.S` from running

Reply:
I've added some checks. I can't use any state from coreclr becuase this code is in PAL. Instead, I've added a function which reads the vector length. In `context.cpp` call the function, if it's not 128bits then it skips all SVE code.

Reply:
Checked this by generating a SIGILL in an SVE test. In debugger, checked the state isn't copied on 256bit, and that it is copied on 128bit